packages/core/src/lib/select/typeahead-input.svelte

```diff
@@ -0,0 +1,149 @@
+<script lang="ts">
+import { theme } from '../../theme';
+import Select from 'svelte-select';
+import type { ComponentProps, ComponentEvents } from 'svelte';
+import Icon from '../icon/icon.svelte';
+import type { IconName } from '../icon/icons';
+
+import { InputStates, type InputState } from '$lib/input';
+import type classNames from 'classnames';
+
+type Events = ComponentEvents<Select>;
+
+interface SelectItem {
+  value: string;
+  label: string;
+  description?: string;
+  icon?: string;
+  group?: string;
+}
+
+type Option = SelectItem | string;
+
+interface $$Props extends Omit<ComponentProps<Select>, 'items'> {
+  value?: string | string[];
+  options: Option[];
+  otherOptionPrefix?: string;
+  state?: InputState;
+  onBlur?: (event: Events['blur']['detail']) => void;
+  onChange?: (event: Events['change']) => void;
+  onClear?: (event: Events['clear']) => void;
+  onFilter?: (event: Events['filter']) => void;
+  onFocus?: (event: Events['focus']) => void;
+  onInput?: (event: Events['focus']) => void;
+  onSelect?: (event: Events['select']) => void;
+}
+
+export let options: $$Props['options'];
+export let inputAttributes: $$Props['inputAttributes'] = undefined;
+export let containerStyles: $$Props['containerStyles'] = '';
+export let state: $$Props['state'] = InputStates.NONE;
+export let onBlur: $$Props['onBlur'] = undefined;
+export let onChange: $$Props['onChange'] = undefined;
+export let onClear: $$Props['onClear'] = undefined;
+export let onFilter: $$Props['onFilter'] = undefined;
+export let onFocus: $$Props['onFocus'] = undefined;
+export let onInput: $$Props['onInput'] = undefined;
+export let onSelect: $$Props['onSelect'] = undefined;
+
+let className = '';
+export { classNames as class };
+
+$: normalizedOptions = options.map((option) => {
+  if (typeof option === 'string') {
+    return { value: option, label: option };
+  }
+
+  return option;
+});
```

Comment:
**thought (performance):** This might be less of a big deal once Svelte 5 is in play, but this list of objects situation has given us a lot of grief in terms of unnecessary DOM operations from this component. I wonder if there's a path here - maybe in a future PR - where we lean more heavily into slots / snippets for stuff like option metadata rather than try to jam it all into props. Seems a little closer to how `svelte-select` itself works